builtin/providers/aws/resource_aws_elasticache_replication_group.go

```diff
@@ -0,0 +1,419 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsElasticacheReplicationGroup() *schema.Resource {
+
+	resourceSchema := resourceAwsElastiCacheCommonSchema()
+
+	resourceSchema["replication_group_id"] = &schema.Schema{
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validateAwsElastiCacheReplicationGroupId,
+	}
+
+	resourceSchema["automatic_failover_enabled"] = &schema.Schema{
+		Type:     schema.TypeBool,
+		Optional: true,
+		Default:  false,
+	}
+
+	resourceSchema["replication_group_description"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Required: true,
+	}
+
+	resourceSchema["number_cache_clusters"] = &schema.Schema{
+		Type:     schema.TypeInt,
+		Required: true,
+		ForceNew: true,
```

Comment:
:-/

You can add read replicas by calling CreateCacheCluster, maybe something for later on: https://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/Replication.AddReadReplica.html#Replication.AddReadReplica.API